src/search/search_encoding.h

```diff
@@ -40,6 +40,8 @@ class IndexMetadata {
   uint8_t flag = 0;  // all reserved
   IndexOnDataType on_data_type;
 
+  const std::string &OnDataTypeString() const { return RedisTypeNames[(size_t)on_data_type]; }
```

Comment:
For me, with the `On` prefix the function sounds like a callback :)

Reply:
Haha make sense. But I think we can now let leave it as is to avoid massive change.